### PR TITLE
WIP: #2937 mailbehaviour charset

### PR DESCRIPTION
**Pull request by ddpardo**
Implementation to support default charset from configuration.

#### Check List:
* Unit tests: YES (partial)
* Documentation: YES (javadoc)

### CODE REVIEW

File: modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/behavior/impl/MailActivityBehavior.java [hunk 1]
```diff
@@ -362,9 +364,14 @@ protected void setEmailSession(Email email, String mailSessionJndi) {
         }
     }
 
-    protected void setCharset(Email email, String charSetStr) {
+    protected void setCharset(Email email, String charSetStr, String tentantId) {
         if (charset != null) {
             email.setCharset(charSetStr);
+        }else{
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/behavior/impl/MailActivityBehavior.java:370]: This is a nitpick but please follow project standards for formatting.  For example spaces before and after braces and parentheses.  Thanks.  

There are formatting files for IntelliJ and Eclipse in https://github.com/flowable/flowable-engine/tree/master/ide-settings

File: modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/behavior/impl/MailActivityBehavior.java [hunk 2]
```diff
@@ -362,9 +364,14 @@ protected void setEmailSession(Email email, String mailSessionJndi) {
         }
     }
 
-    protected void setCharset(Email email, String charSetStr) {
+    protected void setCharset(Email email, String charSetStr, String tentantId) {
```
Comment (filiphr) [inline modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/behavior/impl/MailActivityBehavior.java:367]: Typo in the parameter name. It should be `tenantId`

**Review by filiphr — CHANGES_REQUESTED** [review]
This looks as expected @ddpardo. I've left a comment in one place about a typo.

Please finish the tests for CMMN and the comment given by @dbmalkovsky and we'll merge this.

btw, I see that there is some conflict in the MailActivityBehavior as well, can you please resolve that as well?

**Comment by filiphr** [comment]
Thanks for the PR @ddpardo. Since we are closing to our release, I took this and finished it

**Pull request closed without merging**